src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -260,7 +260,12 @@ protected Object invokeStep(StepDescriptor d, String name, Object args) {
             }
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
-            s = d.newInstance(ps.namedArgs);
+            if (Util.isOverridden(StepDescriptor.class, d.getClass(), "newInstance", Map.class)) {
```

Comment:
…which is deprecated, so we do not care too much about those cases.

Reply:
Deprecated yes, but in particular this case will always used by `ParallelStep`, and what `ParallelStep` is doing cannot currently be replicated with `CustomDescribableModel`, so it is still somewhat important.

On the other hand overrides like [this](https://github.com/jenkinsci/pipeline-utility-steps-plugin/blob/0b912dac3870b880d354dcbe0d21cfff4b08ff93/src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/AbstractFileOrTextStepDescriptorImpl.java#L24) could definitely be replaced with `CustomDescribableModel`.

Reply:
> overrides like [this](https://github.com/jenkinsci/pipeline-utility-steps-plugin/blob/0b912dac3870b880d354dcbe0d21cfff4b08ff93/src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/AbstractFileOrTextStepDescriptorImpl.java#L24) could definitely be replaced with `CustomDescribableModel`

Better to delete the override and throw `AbortException` from `StepExecution.start`.